### PR TITLE
Create Location Services utility

### LINKED ISSUE #93: Request Location Services Permission Utility
* Include ability to ask for Location OR report is user has already denied access 
* Alert needs to include a reason description (configured in the Info.plist)

Screen Shot 2020-05-17 at 4 13 07 PM

### PR DESCRIPTION
**Pull request by haidalinda**
**What this PR does**:

This PR creates a utility (`LocationServices`) that wraps all interaction with `CoreLocation`. It allows clients to request Locations permissions as well as register to observer that status changing.

### Checklist

- [x] Compiler warnings resolved
- [x] Linter warnings resolved
- [x] User-facing strings in Localizable.strings file
- [x] Unused code, print statements, and comments removed

**Which issue(s) this PR fixes**:

Resolves #93 

**Special notes for reviewers**:

See in action by selecting the Filters nav bar button and viewing the Filters modal

If you select `Deny`, you'll have to elect to `Erase All Content and Settings` from the Simulator to get back to a state where permissions are requestable again

**Additional comments**:

Observation pattern implementation from: https://www.swiftbysundell.com/articles/observers-in-swift-part-1/

### CODE REVIEW

File: FightPandemics/Features/Filters/FiltersModal.swift [hunk 1]
```diff
@@ -27,22 +27,52 @@
...
+
     override func viewDidLoad() {
         super.viewDidLoad()
 
         setupUI()
+        locationServices.addObserver(self)
+        locationServices.requestPermission()
     }
 
+    // MARK: - Instance mtehods
```
Comment (stavares843) [inline FightPandemics/Features/Filters/FiltersModal.swift:55]: typo - methods

**Review by stavares843 — COMMENTED** [review]
- should we add more spacing on the copy and add a button for the user to go directly to the settings device to update the permissions? https://jmp.sh/eLIAV7S
- if you go to the filters > allow once > dismiss modal > filters again > appears like this - https://jmp.sh/DWu2AOt 
- when you select don't allow > delete app > install again > doesn't ask again the permission

**Comment by haidalinda** [comment]
> should we add more spacing on the copy and add a button for the user to go directly to the settings device to update the permissions? https://jmp.sh/eLIAV7S

the copy is just a placeholder - for when we have actual specs for what to display when permissions are denied; that's out of scope of this Issue, IMO

> if you go to the filters > allow once > dismiss modal > filters again > appears like this - https://jmp.sh/DWu2AOt

good catch! i'll look into that

> when you select don't allow > delete app > install again > doesn't ask again the permission

that's unfortunately not something we can control!

**Comment by stavares843** [comment]
humm

strange, on previous iOS versions would ask again the permission or its only on iOS 13?

**Comment by haidalinda** [comment]
> strange, on previous iOS versions would ask again the permission or its only on iOS 13?

it would seem so - there's nothing i can see with the framework we can/should do to control that 🤷

**Comment by stavares843** [comment]
> if you go to the filters > allow once > dismiss modal > filters again > appears like this - jmp.sh/DWu2AOt

still happens

**Review by ethanswift — APPROVED** [review]

**Review by OnyekachiSamuel — COMMENTED** [review]
This looks good, nice work.

**Review by OnyekachiSamuel — APPROVED** [review]

**Pull request merged**